std/evmprecompiles/01-ecrecover.go

```diff
@@ -0,0 +1,79 @@
+package evmprecompiles
+
+import (
+	"fmt"
+
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/algebra/emulated/sw_emulated"
+	"github.com/consensys/gnark/std/math/bits"
+	"github.com/consensys/gnark/std/math/emulated"
+)
+
+// ECRecover implements [ECRECOVER] precompile contract at address 0x01.
+//
+// [ECRECOVER]: https://ethereum.github.io/execution-specs/autoapi/ethereum/paris/vm/precompiled_contracts/ecrecover/index.html
+func ECRecover(api frontend.API, msg emulated.Element[emulated.Secp256k1Fr],
```

Comment:
maybe add ECRecover function to `std/signatures/ecdsa` as in gnark-crypto and use function here? but tbh the only use of ECDSA I can see is Ethereum.

Reply:
I tried and it almost works until I got to hints used in ECrecover. The hints cannot be generic and thus I have to pass all the field emulation parameters (modulus, nb limbs, limb width) into the hint. Then, I would have to switch between different gnark-crypto `ECrecover` methods (we have in `bn254` and `secp256k1` package) depending on the modulus. It got quite messy.

I would rather postpone and keep it as an issue for future.